components/webext-storage/src/migration.rs

```diff
@@ -198,7 +286,30 @@ fn do_insert(tx: &Transaction<'_>, ext_id: &str, vals: Vec<(String, Value)>) ->
             ":data": &Value::Object(map),
         },
     )?;
-    Ok(())
+    Ok(num_entries)
+}
+
+#[derive(Debug, Clone, Default, PartialEq)]
+pub struct MigrationInfo {
+    /// The number of entries (rows in the original table) we attempted to
+    /// migrate. Zero if there was some error in computing this number.
+    pub entries: usize,
+    /// The number of records we successfully migrated (equal to `entries` for
+    /// entirely successful migrations).
+    pub entries_successful: usize,
+    /// The number of extensions (distinct extension ids) in the original
+    /// table.
+    pub extensions: usize,
+    /// The number of extensions we successfully migrated
+    pub extensions_successful: usize,
+    /// A count of recoverable non-fatal errors seen. Generally this will cause
+    /// the failed count to be nonzero, but there are other cases that can cause
+    /// error (failing to compute # of entries or extensions).
+    pub non_fatal_errors_seen: usize,
+    /// True if we hit at least one quota error.
+    pub overflowed_quotas: bool,
```

Comment:
So a quota error is not a `non_fatal_error`, right? But it also means it's not a successful migration (so no +1 to `extensions_successful`)—or does it? We did technically insert it into the DB, even though the extension can't do anything with it, because it won't be able to make any more updates without running into quota errors.

I guess I'd love to discuss more about how all these counts overlap, and don't, and what kind of questions we'll want to answer in analysis.